packages/core/src/App/Containers/Layout/app-contents.jsx

```diff
@@ -32,28 +31,33 @@ const AppContents = ({
     const [show_cookie_banner, setShowCookieBanner] = React.useState(false);
     const [is_gtm_tracking, setIsGtmTracking] = React.useState(false);
     const { is_appstore } = React.useContext(PlatformContext);
-    const {
-        client: { user_id },
-    } = useStore();
 
     const tracking_status = tracking_status_cookie.get(TRACKING_STATUS_KEY);
 
     const scroll_ref = React.useRef(null);
 
+    const current_page = window.location.hostname + window.location.pathname;
+
     React.useEffect(() => {
         // rudderstack page view trigger
-        WS.wait('authorize').then(() => {
-            RudderStack.identifyEvent(user_id, {
-                language: getLanguage().toLowerCase() || 'en',
-            });
-            const current_page = window.location.hostname + window.location.pathname;
-            RudderStack.pageView(current_page);
+        WS.wait('authorize').then(response => {
+            if (response.error) return;
+            const user_id = response.authorize?.user_id;
```

Comment:
Check user_id using optional chaining, destructuring this will yield the error message:
<img width="449" alt="Screenshot 2023-07-18 at 6 39 04 PM" src="https://github.com/binary-com/deriv-app/assets/103016120/91597564-6faa-48d1-a8fa-e03f5ef25ee5">